From 098d4ea15acd0c63df70723dcb647baa7e5ebc1e Mon Sep 17 00:00:00 2001
From: Sekhar Nori <nsekhar@ti.com>
Date: Sun, 24 Nov 2013 20:43:08 +0530
Subject: [PATCH 171/285] ARM: dts: am437x-gp-evm: add support for clkout2

Provide support for clkout2 pin on AM437x GP EVM.
On this EVM, this pin provides the 32khz clock to
WLAN module.

Signed-off-by: Sekhar Nori <nsekhar@ti.com>
Signed-off-by: Kishon Vijay Abraham I <kishon@ti.com>
[Kevin: The original patch taken from TI
ti-sdk-am437x-evm-07.01.00.00-Linux-x86-Install.bin]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/boot/dts/am437x-gp-evm.dts |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/arch/arm/boot/dts/am437x-gp-evm.dts b/arch/arm/boot/dts/am437x-gp-evm.dts
index 8e248c7..b6271dc 100644
--- a/arch/arm/boot/dts/am437x-gp-evm.dts
+++ b/arch/arm/boot/dts/am437x-gp-evm.dts
@@ -112,7 +112,8 @@
 };
 
 &am43xx_pinmux {
-
+	pinctrl-names = "default";
+	pinctrl-0 = <&dss_pinctrl &clkout2_pin>;
 	cpsw_default: cpsw_default {
 		pinctrl-single,pins = <
 			/* Slave 1 */
@@ -246,6 +247,12 @@
 		>;
 	};
 
+	clkout2_pin: clkout2_pin {
+		pinctrl-single,pins = <
+			0x274 (PIN_OUTPUT_PULLUP | MUX_MODE3)
+		>;
+	};
+
 	dcan0_default: dcan0_default_pins {
 		pinctrl-single,pins = <
 			0x178 (PIN_OUTPUT_PULLDOWN | MUX_MODE2) /* uart1_ctsn.d_can0_tx */
@@ -694,3 +701,7 @@
 		};
 	};
 };
+
+&clkout_32k_mux_ck {
+	clock-default = <&clk_32768_ck>;
+};
-- 
1.7.5.4

